docs/courses/csintro1/functions/extensions.md

```diff
@@ -0,0 +1,186 @@
+
+In this activity, students will:
+* Import an extension
+* Write a small extension
+* Share their extension with other students
+
+## Using Shared Code
+
+Now that we have experience writing functions, we might notice something different about the blocks you've been using - almost all of them are already functions! These functions are provided to you by default to make your coding experience easier. We call these "built in functions."
+
+We are not stuck with only the functions provided when you first open MakeCode. Extensions, in the advanced menu, allow functions, and other code, to be included and used in our games. 
+
+We can make even make our own packages of code to share or reuse in our games.
+
+## Example #1: Using a package
+
+https://youtu.be/VVjxMS29jjU
+
+[Alternative Video Location](https://aka.ms/40544a-extensiondart) 
+
+1. Open a new project (name it "football")
+2. Using the extensions menu, search for "darts" (or the full package location at [github.com/jwunderl/pxt-darts](github.com/jwunderl/pxt-darts)). Notice how a new section is added to the menu, above ``||game:Game||``
+3. Review the example code below, and either recreate it or build something similar
+
+![Finding Dart Extension](/static/courses/csintro1/functions/finding-darts.gif)
+
+```blocks
+enum SpriteKind {
+    Player,
+    Enemy,
+    Foot_Ball,
+    Goal
+}
+let mySprite: Sprite = null
+let myDart: Dart = null
+controller.A.onEvent(ControllerButtonEvent.Pressed, function () {
+    myDart.throwDart()
+})
+sprites.onOverlap(SpriteKind.Foot_Ball, SpriteKind.Goal, function (sprite, otherSprite) {
+    game.splash("Goal!")
+})
+myDart = Darts.create(img`
+. . . . . . . . . . . . . . . . 
+. . . . . . . . . . . . . . . . 
+. . . . . . . . . e e e . . . . 
+. . . . . . e e e e d e e . . . 
+. . . . . e e d d d d d e e . . 
+. . . . e e d d d d d d d e . . 
+. . . . e d d d d d d d d e . . 
+. . . e e d d d d d d d d e . . 
+. . . e d d d d d d d d d e . . 
+. . . e d d d d d d d d d e . . 
+. . . e e e d d d d d e e e . . 
+. . . . . e e e e e e e . . . . 
+. . . . . . . . . . . . . . . . 
+. . . . . . . . . . . . . . . . 
+. . . . . . . . . . . . . . . . 
+. . . . . . . . . . . . . . . . 
+`, SpriteKind.Foot_Ball)
+myDart.setTrace()
+myDart.controlWithArrowKeys()
+mySprite = sprites.create(img`
+. . . . . . . . . . . . . . . . 
+. . 1 1 1 1 1 1 1 1 1 1 1 1 . . 
+. . . . . . . 1 . . . . . 1 . . 
+. . . . . . . . 1 . . . . 1 . . 
+. . . . . . . . . 1 . . . 1 . . 
+. . . . . . . . 1 . 1 . . 1 . . 
+. . . . . . . 1 . . . 1 . 1 . . 
+. . . . . . . 1 . . . . 1 1 . . 
+. . . . . . . 1 1 1 1 1 1 1 . . 
+. . . . . . . 1 . . . . 1 1 . . 
+. . . . . . . 1 . . . 1 . 1 . . 
+. . . . . . . . 1 . 1 . . 1 . . 
+. . . . . . . . . 1 . . . 1 . . 
+. . . . . . . . 1 . . . . 1 . . 
+. . . . . . . 1 . . . . . 1 . . 
+. . . 1 1 1 1 1 1 1 1 1 1 1 . . 
+`, SpriteKind.Goal)
+mySprite.setPosition(150, 110)
+```
+
+```package
+darts=github:jwunderl/pxt-darts#master
```

Comment:
Just to confirm, this would be done by switching this line to

`darts=github:jwunderl/pxt-darts#v0.0.14`

to switch to the release tagged v0.0.14, correct?

Reply:
yes; I think you have another `#master` somewhere

Reply:
in commit 91bd68f I changed it in the three locations that extensions are used. Thanks!